### PR TITLE
[zh-cn] init translation of html attribute required

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

part of #17632

### CODE REVIEW

File: files/zh-cn/web/html/attributes/required/index.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+slug: Web/HTML/Attributes/required
+l10n:
+  sourceCommit: aee2bd82de11cb7331134e48e8bd548bbedafcc5
+---
+
+{{HTMLSidebar}}
+
+**`required`** [布尔](/zh-CN/docs/Glossary/Boolean/HTML)属性（如果存在）表示用户在提交自有表单前必须为输入指定一个值。
+
+`{{HTMLElement("input/text", "text")}}`、`{{HTMLElement("input/search", "search")}}`、`{{HTMLElement("input/url", "url")}}`、`{{HTMLElement("input/tel", "tel")}}`、`{{HTMLElement("input/email", "email")}}`、 `{{HTMLElement("input/password", "password")}}`、`{{HTMLElement("input/date", "date")}}`、`{{HTMLElement("input/month", "month")}}`、`{{HTMLElement("input/week", "week"）}}`、`{{HTMLElement("input/time", "time")}}`、`{{HTMLElement("input/datetime-local", "datetime-local")}}`、`{{HTMLElement("input/number", "number")}}`、`{{HTMLElement("input/checkbox", "checkbox")}}`、`{{HTMLElement("input/radio", "radio")}}`、{{HTMLElement("input/file", "file")}} {{HTMLElement("input")}} 类型以及 {{HTMLElement("select")}} 和 {{HTMLElement("textarea")}} 表单控制元素均支持 `required` 属性。如果这些输入类型和元素中出现了 {{cssxref(':required')}} 伪类，则将与之匹配。如果不包含该属性，则将匹配 {{cssxref(':optional')}} 伪类。
```
Comment (jasonren0403) [inline files/zh-cn/web/html/attributes/required/index.md:12]:
```suggestion
`{{HTMLElement("input/text", "text")}}`、`{{HTMLElement("input/search", "search")}}`、`{{HTMLElement("input/url", "url")}}`、`{{HTMLElement("input/tel", "tel")}}`、`{{HTMLElement("input/email", "email")}}`、 `{{HTMLElement("input/password", "password")}}`、`{{HTMLElement("input/date", "date")}}`、`{{HTMLElement("input/month", "month")}}`、`{{HTMLElement("input/week", "week"）}}`、`{{HTMLElement("input/time", "time")}}`、`{{HTMLElement("input/datetime-local", "datetime-local")}}`、`{{HTMLElement("input/number", "number")}}`、`{{HTMLElement("input/checkbox", "checkbox")}}`、`{{HTMLElement("input/radio", "radio")}}`、`{{HTMLElement("input/file", "file")}}` {{HTMLElement("input")}} 类型以及 {{HTMLElement("select")}} 和 {{HTMLElement("textarea")}} 表单控制元素均支持 `required` 属性。如果这些输入类型和元素中出现了 {{cssxref(':required')}} 伪类，则将与之匹配。如果不包含该属性，则将匹配 {{cssxref(':optional')}} 伪类。
```

File: files/zh-cn/web/html/attributes/required/index.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+slug: Web/HTML/Attributes/required
+l10n:
+  sourceCommit: aee2bd82de11cb7331134e48e8bd548bbedafcc5
+---
+
+{{HTMLSidebar}}
+
+**`required`** [布尔](/zh-CN/docs/Glossary/Boolean/HTML)属性（如果存在）表示用户在提交自有表单前必须为输入指定一个值。
+
+`{{HTMLElement("input/text", "text")}}`、`{{HTMLElement("input/search", "search")}}`、`{{HTMLElement("input/url", "url")}}`、`{{HTMLElement("input/tel", "tel")}}`、`{{HTMLElement("input/email", "email")}}`、 `{{HTMLElement("input/password", "password")}}`、`{{HTMLElement("input/date", "date")}}`、`{{HTMLElement("input/month", "month")}}`、`{{HTMLElement("input/week", "week"）}}`、`{{HTMLElement("input/time", "time")}}`、`{{HTMLElement("input/datetime-local", "datetime-local")}}`、`{{HTMLElement("input/number", "number")}}`、`{{HTMLElement("input/checkbox", "checkbox")}}`、`{{HTMLElement("input/radio", "radio")}}`、`{{HTMLElement("input/file", "file")}}` {{HTMLElement("input")}} 类型以及 {{HTMLElement("select")}} 和 {{HTMLElement("textarea")}} 表单控制元素均支持 `required` 属性。如果这些输入类型和元素中出现了 {{cssxref(':required')}} 伪类，则将与之匹配。如果不包含该属性，则将匹配 {{cssxref(':optional')}} 伪类。
```
Comment (jasonren0403) [inline files/zh-cn/web/html/attributes/required/index.md:12]:
```suggestion
`{{HTMLElement("input/text", "text")}}`、`{{HTMLElement("input/search", "search")}}`、`{{HTMLElement("input/url", "url")}}`、`{{HTMLElement("input/tel", "tel")}}`、`{{HTMLElement("input/email", "email")}}`、`{{HTMLElement("input/password", "password")}}`、`{{HTMLElement("input/date", "date")}}`、`{{HTMLElement("input/month", "month")}}`、`{{HTMLElement("input/week", "week")}}`、`{{HTMLElement("input/time", "time")}}`、`{{HTMLElement("input/datetime-local", "datetime-local")}}`、`{{HTMLElement("input/number", "number")}}`、`{{HTMLElement("input/checkbox", "checkbox")}}`、`{{HTMLElement("input/radio", "radio")}}`、`{{HTMLElement("input/file", "file")}}` {{HTMLElement("input")}} 类型以及 {{HTMLElement("select")}} 和 {{HTMLElement("textarea")}} 表单控制元素均支持 `required` 属性。如果这些输入类型和元素中出现了 {{cssxref(':required')}} 伪类，则将与之匹配。如果不包含该属性，则将匹配 {{cssxref(':optional')}} 伪类。
```

File: files/zh-cn/web/html/attributes/required/index.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+---
+title: HTML 属性：required
+slug: Web/HTML/Attributes/required
+l10n:
+  sourceCommit: aee2bd82de11cb7331134e48e8bd548bbedafcc5
+---
+
+{{HTMLSidebar}}
+
+**`required`** [布尔](/zh-CN/docs/Glossary/Boolean/HTML)属性（如果存在）表示用户在提交自有表单前必须为输入指定一个值。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/required/index.md:10]:
```suggestion
**`required`** [布尔](/zh-CN/docs/Glossary/Boolean/HTML)属性（如果存在）表示用户在提交输入所属的表单前必须为其指定一个值。
```

File: files/zh-cn/web/html/attributes/required/index.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+`{{HTMLElement("input/text", "text")}}`、`{{HTMLElement("input/search", "search")}}`、`{{HTMLElement("input/url", "url")}}`、`{{HTMLElement("input/tel", "tel")}}`、`{{HTMLElement("input/email", "email")}}`、`{{HTMLElement("input/password", "password")}}`、`{{HTMLElement("input/date", "date")}}`、`{{HTMLElement("input/month", "month")}}`、`{{HTMLElement("input/week", "week")}}`、`{{HTMLElement("input/time", "time")}}`、`{{HTMLElement("input/datetime-local", "datetime-local")}}`、`{{HTMLElement("input/number", "number")}}`、`{{HTMLElement("input/checkbox", "checkbox")}}`、`{{HTMLElement("input/radio", "radio")}}`、`{{HTMLElement("input/file", "file")}}` {{HTMLElement("input")}} 类型以及 {{HTMLElement("select")}} 和 {{HTMLElement("textarea")}} 表单控制元素均支持 `required` 属性。如果这些输入类型和元素中出现了 {{cssxref(':required')}} 伪类，则将与之匹配。如果不包含该属性，则将匹配 {{cssxref(':optional')}} 伪类。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/required/index.md:12]:
```suggestion
`{{HTMLElement("input/text", "text")}}`、`{{HTMLElement("input/search", "search")}}`、`{{HTMLElement("input/url", "url")}}`、`{{HTMLElement("input/tel", "tel")}}`、`{{HTMLElement("input/email", "email")}}`、`{{HTMLElement("input/password", "password")}}`、`{{HTMLElement("input/date", "date")}}`、`{{HTMLElement("input/month", "month")}}`、`{{HTMLElement("input/week", "week")}}`、`{{HTMLElement("input/time", "time")}}`、`{{HTMLElement("input/datetime-local", "datetime-local")}}`、`{{HTMLElement("input/number", "number")}}`、`{{HTMLElement("input/checkbox", "checkbox")}}`、`{{HTMLElement("input/radio", "radio")}}`、`{{HTMLElement("input/file", "file")}}` {{HTMLElement("input")}} 类型以及 {{HTMLElement("select")}} 和 {{HTMLElement("textarea")}} 表单控件元素均支持 `required` 属性。如果这些输入类型和元素中出现了该属性，则会匹配 {{cssxref(':required')}} 伪类。如果不包含该属性，则会匹配 {{cssxref(':optional')}} 伪类。
```

File: files/zh-cn/web/html/attributes/required/index.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+{{HTMLElement("input/range", "range")}} 和 {{HTMLElement("input/color", "color")}} 不支持该属性，也与之无关，因为两者都有默认值。{{HTMLElement("input/hidden", "hidden")}} 也不支持该属性，因为用户不可能填写隐藏的表单。也不支持任何按钮类型，包括 `image`。
+
+注意，`color` 和 `range` 也不支持 `required`，因为它们都总是有一个值。`color` 类型默认为 `#000000`。`range` 的默认值是 `min` 和 `max` 之间的中点，如果没有声明，在大多数浏览器中，`min` 和 `max` 的默认值分别是 0 和 100。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/required/index.md:16]: Somewhat duplicated with the previous paragraph, could you fix it in mdn/content first :)

File: files/zh-cn/web/html/attributes/required/index.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+对于由 {{HTMLElement("input/radio", "radio")}} 按钮组成的同名组，如果该组中的某个单选按钮具有 `required` 属性，则必须选中该组中的某个单选按钮，但该单选按钮不一定是应用了该属性的单选按钮。为改善代码维护，建议在组中的每个同名单选按钮中都包含 `required` 属性，否则一个都不包含。
+
+如果是同名的一组 {{HTMLElement("input/checkbox", "checkbox")}} 输入类型，则只有带有 `required` 属性的复选框是必填的。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/required/index.md:20]: Check box is not using "填充"

```suggestion
如果是同名的一组 {{HTMLElement("input/checkbox", "checkbox")}} 输入类型，则只有带有 `required` 属性的复选框是必选的。
```

File: files/zh-cn/web/html/attributes/required/index.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+在包含 `required` 属性时，应在控件附近提供可见提示，告知用户 {{HTMLElement("input")}}、{{HTMLElement("select")}} 或 {{HTMLElement("textarea")}} 为必填项。此外，使用 {{cssxref(':required')}} 伪类来定位必填表单控件，并对其进行样式设计，以表明它们是必填的。这可以提高视力正常用户的可用性。辅助技术应根据 required 属性告知用户表单控件是必填的，但添加 `aria-required="true"` 也无妨，以避免浏览器/屏幕阅读器组合还不支持 `required`。
+
+### 约束验证
+
+如果元素为必填元素，且元素值为空字符串，则该元素会受到 {{domxref('ValidityState.valueMissing','valueMissing')}} 的影响，元素将匹配 {{cssxref(':invalid')}} 伪类。
+
+## 无障碍考虑
+
+向用户提供提示，告知他们需要使用表单控件。确保信息传递是多方面的，例如通过文字、颜色、标记和属性，这样，无论用户是色盲、存在认知差异还是使用了屏幕阅读器，都能理解相关要求。
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/required/index.md:39]:
```suggestion
向用户提供提示，告知他们特定表单控件是必填的。确保信息传递是多方面的，例如通过文字、颜色、标记和属性，这样，无论用户是色盲、存在认知差异还是使用了屏幕阅读器，都能理解相关要求。
```

File: files/zh-cn/web/html/attributes/required/index.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  </div>
+  <div class="group">
+    <input type="text" required />
+    <label>必需</label>
+  </div>
+  <input type="submit" />
+</form>
+```
+
+### 运行结果
```
Comment (yin1999) [inline files/zh-cn/web/html/attributes/required/index.md:59]:
```suggestion
### 结果
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions:

File: files/zh-cn/web/html/attributes/required/index.md (hunk 5 above)
Comment (jasonren0403) [reply files/zh-cn/web/html/attributes/required/index.md:16]: > Somewhat duplicated with the previous paragraph, could you fix it in mdn/content first :)

proposed https://github.com/mdn/content/pull/35318

**Review by yin1999 — APPROVED** [review]

**Pull request merged**